### ISSUE TITLE
How to read some values reported by pgmetrics (e.g. shared_buffers, max_wal_size)

### ISSUE THREAD

**Issue by romulusFR** [rapidloop/pgmetrics#34]
Hello. I think i've an issue similar to #2 Command `SHOW shared_buffers;` shows

```
 shared_buffers 
----------------
 4GB
(1 row)
```

But pgmetrics reports the following value 524288. 

```
    +---------------------------------+-------------------+
    |                         Setting |             Value |
    +---------------------------------+-------------------+
    |                  shared_buffers |  524288 (4.0 GiB) |
    |                        work_mem |    32768 (32 MiB) |
    |            maintenance_work_mem | 1048576 (1.0 GiB) |
    |                    temp_buffers |    1024 (8.0 MiB) |
```

What are the units and the values on the left ?

Similarly, and more inconsistently  `SHOW max_wal_size;` returns
```
 max_wal_size 
--------------
 2GB
(1 row)
```
But pgmetrics reports as follows

```
    +--------------------+---------------+
    |            Setting |         Value |
    +--------------------+---------------+
    |          wal_level |       replica |
    |    archive_timeout |             0 |
    |    wal_compression |           off |
    |       max_wal_size | 2048 (32 GiB) |
    |       min_wal_size | 512 (8.0 GiB) |
    | checkpoint_timeout |           300 |
    |   full_page_writes |            on |
    |  wal_keep_segments |             0 |
    +--------------------+---------------+


```

**Comment by mchristofides** [comment]
Saw this and realised I could help.

`shared_buffers`  without units is specified in blocks, typically 8kB – [docs](https://www.postgresql.org/docs/current/runtime-config-resource.html)

`max_wal_size`  without units is specified in segment files, typically 16MB – [docs](https://www.postgresql.org/docs/current/continuous-archiving.html#BACKUP-ARCHIVING-WAL)

**Comment by mdevan** [comment]
Yes, @mchristofides is correct. pgmetrics is showing the literal value in defined units of the setting, and in paranthesis, the effective value in bytes. In the line:

```
|                  shared_buffers |  524288 (4.0 GiB) |
```

it is trying to say that the value is 524288 blocks, which means 4.0 GB since each block is of size 8192 bytes.

**Comment by romulusFR** [comment]
Thanks @mchristofides and @mdevan. So I'm OK with `shared_buffers` as _524288 blocks (4.0 GiB)_ is consistent with my settings. Maybe a default `blocks` unit would be worth but pgmetrics is ok, let's say it's and enhancement request.

Now, I understand `max_wal_size` and pgmetrics seems wrong on the left figure misinterpreting the default unit as 16MB, one should read _2048 (MB) (4 GiB)_ and not _2048 (16MB) (32 GiB)_ to  be consistent with my settings. In that case it's closer to a bug maybe.

**Comment by mdevan** [comment]
Hmm, yes, you're right, there is a bug here.

For Postgres 9.4 and below, the setting was called `checkpoint_segments` (as the number of segments, unitless). For 9.5 and 9.6 the setting is `max_wal_size` with a unit of 16MB and for 10 and above it is `max_wal_size` with a unit of 1MB.

pgmetrics reporting code assumes 16MB for 9.5+, it does not use 1MB for 10+. Are you using Postgres 10 or above?

Will fix.

**Comment by romulusFR** [comment]
For this specific case, it's PostgreSQL 12 indeed.

**Comment by mdevan** [comment]
Fixed in 513dcc0. Thanks for reporting.
